src/sentry/api/endpoints/organization_derive_code_mappings.py

```diff
@@ -0,0 +1,116 @@
+from typing import List
+
+from rest_framework import status
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry import features
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases.organization import (
+    OrganizationEndpoint,
+    OrganizationIntegrationsLoosePermission,
+)
+from sentry.api.serializers import serialize
+from sentry.integrations.utils.code_mapping import (
+    CodeMappingMatch,
+    CodeMappingTreesHelper,
+    FrameFilename,
+)
+from sentry.models import Project, Repository, RepositoryProjectPathConfig
+from sentry.models.organization import Organization
+from sentry.tasks.derive_code_mappings import get_installation
+
+
+@region_silo_endpoint
+class OrganizationDeriveCodeMappingsEndpoint(OrganizationEndpoint):
+    permission_classes = (OrganizationIntegrationsLoosePermission,)
+
+    def get(self, request: Request, organization: Organization) -> Response:
+        """
+        Get all matches for a filename.
+        ``````````````````
+
+        :pparam string organization_slug: the slug of the organization the
+                                          team should be created for.
+        :param string filename:
+        :auth: required
+        """
+        if not features.has("organizations:derive-code-mappings", organization):
+            return Response(status=status.HTTP_400_BAD_REQUEST)
+
+        filename = request.GET.get("filename")
+        installation, _ = get_installation(organization)
+        if not installation:
+            return self.respond(
+                "Could not find this integration installed on your organization",
+                status=status.HTTP_404_NOT_FOUND,
+            )
+
+        trees = installation.get_trees_for_org()
+        trees_helper = CodeMappingTreesHelper(trees)
+        frame_filename = FrameFilename(filename)
+        possible_code_mappings: List[CodeMappingMatch] = trees_helper.list_file_matches(
```

Comment:
Instead of calling `generate_code_mappings`, we reuse some of that logic to get all the files that could be a possible match. 

We can include the information needed to create a code mapping and count on that for the post request.